src/github.com/getlantern/flashlight/client/pluggable_transports.go

```diff
@@ -82,10 +87,21 @@ func obfs4DialFactory(s *ChainedServerInfo, deviceID string) (dialFN, error) {
 
 	args, err := cf.ParseArgs(ptArgs)
 	if err != nil {
-		return nil, fmt.Errorf("Unable to parse client args: %v", err)
+		return nil, log.Errorf("Unable to parse client args: %v", err)
 	}
 
 	return func() (net.Conn, error) {
-		return cf.Dial("tcp", s.Addr, net.Dial, args)
+		defer proxyContext(s.Addr, "obfs4").Exit()
+		conn, err := cf.Dial("tcp", s.Addr, net.Dial, args)
+		return conn, log.IfError(err)
 	}, nil
 }
+
+func proxyContext(addr string, protocol string) *context.Context {
+	ctx := context.Enter().Put("proxy_protocol", protocol)
```

Comment:
I see, you need to do it in the calls to proxyContext(). However, I find this unintuitive and potentially prone to developer errors, as it seems an inconsistency. What's the origin of this?